app/api/events.py

```diff
@@ -385,6 +398,19 @@ def before_get_object(self, view_kwargs):
             else:
                 view_kwargs['id'] = None
 
+    def before_patch(self, args, kwargs, data=None):
+        """
+        before patch method to verify if the event location is provided before publishing the event
+        :param args:
+        :param kwargs:
+        :param data:
+        :return:
+        """
+        if 'state' in data and data['state'] == 'published':
```

Comment:
Looks good, but is it appropriate to do the same thing in the outer `if` that you did in the inner, like:
```
if data.get('state', None) == 'published':
    ...
```

Reply:
Yes, thanks. Updating .....